src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2664,6 +2664,41 @@ internal EventCacheEntry(EventInfo[] events)
             {
                 this.events = events;
             }
+
+            private string _eventDefinition;
+            internal string EventDefinition
+            {
+                get
+                {
+                    if (_eventDefinition == null)
+                    {
+                        var sb = new StringBuilder();
+                        var invokeMethod = new MethodInformation(
+                            events[0].EventHandlerType.GetMethod(nameof(Action.Invoke)), parametersToIgnore: 0).method;
+                        sb.AppendFormat(
+                            CultureInfo.InvariantCulture,
+                            "event {0} {1}",
+                            ToStringCodeMethods.Type(events[0].EventHandlerType),
+                            events[0].Name);
+
+                        sb.Append("(");
+                        sb.AppendJoin(
+                            ", ",
+                            invokeMethod.GetParameters().Select(
+                                p => string.Format(
+                                    CultureInfo.InvariantCulture,
+                                    "{0} {1}",
+                                    ToStringCodeMethods.Type(p.ParameterType),
+                                    p.Name)));
+                        sb.Append(")");
+
+                        _eventDefinition = sb.ToString();
+                        sb.Clear();
```

Comment:
This probably isn't needed -- it's probably simpler to let the GC just collect the whole string builder in one go when the method returns